File: ddpoliglot_v6_flutter/lib/models/data/WordPhraseWord.dart
// ignore_for_file: file_names
import 'Word.dart';
import './WordPhrase.dart';

import 'package:json_annotation/json_annotation.dart';

part 'WordPhraseWord.g.dart';

@JsonSerializable()
class WordPhraseWord {
  int wordPhraseWordID;
  int wordID;
  Word? word;
  int wordPhraseID;
  WordPhrase? wordPhrase;
  int phraseOrder;

  WordPhraseWord(
    this.wordPhraseWordID,
    this.wordID,
    this.word,
    this.wordPhraseID,
    this.wordPhrase,
    this.phraseOrder,
  );

  factory WordPhraseWord.fromJson(Map<String, dynamic> data) =>
      _$WordPhraseWordFromJson(data);

  Map<String, dynamic> toJson() => _$WordPhraseWordToJson(this);
}

File: ddPoliglotV6.Data/Models/WordPhraseWord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ddPoliglotV6.Data.Models
{
    public class WordPhraseWord
    {
        [Key]
        public int WordPhraseWordID { get; set; }

        public int WordID { get; set; }
        [ForeignKey("WordID")]
        public Word Word { get; set; }

        public int WordPhraseID { get; set; }
        [ForeignKey("WordPhraseID")]
        public WordPhrase WordPhrase { get; set; }

        public int PhraseOrder { get; set; }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
